x-pack/legacy/plugins/ml/public/data_frame_analytics/common/analytics.ts

```diff
@@ -189,26 +199,32 @@ export const loadEvalData = async ({
   isTraining,
   index,
   dependentVariable,
+  resultsField,
+  predictionFieldName,
 }: {
   isTraining: boolean;
   index: string;
   dependentVariable: string;
+  resultsField: string;
+  predictionFieldName?: string;
 }) => {
   const results: LoadEvaluateResult = { success: false, eval: null, error: null };
+  const defaultPredictionField = `${dependentVariable}_prediction`;
+  const predictedField = `${resultsField}.${
+    predictionFieldName ? predictionFieldName : defaultPredictionField
+  }`;
+
+  const query = { term: {} };
+  // @ts-ignore TODO: fix types here
+  query.term[`${resultsField}.is_training`] = { value: isTraining };
```

Comment:
You could try to do the original assignment with the computed property name:

```js
const query = { term: { [`${resultsField}.is_training`]: { value: isTraining } } }'
```
```

Reply:
Nice suggestion! Updated in 695aaeda022ae742baf164bbff52271069ee2b0b